frontend/ui/uimanager.lua

```diff
@@ -47,8 +48,42 @@ function UIManager:init()
         -- resume.
         self:_initAutoSuspend()
         self.event_handlers["Suspend"] = function(input_event)
-            self:_stopAutoSuspend()
-            Device:onPowerEvent(input_event)
+            if not self._power_off_triggered then
```

Comment:
Hmm, I've realised there's a flaw in that plan: the power off dialog should show up while the power button is being pressed, otherwise it's going to be annoying to figure out if you've held it for long enough before releasing it. I think we still have to use `UIManager:scheduleIn()` and either `_power_off_triggered` or temporarily removing the `Suspend` handler in this case.